crates/bevy_sprite/src/texture_atlas.rs

```diff
@@ -19,7 +19,7 @@ use bevy_utils::HashMap;
 #[reflect(Debug)]
 pub struct TextureAtlasLayout {
     // TODO: add support to Uniforms derive to write dimensions and sprites to the same buffer
-    pub size: Vec2,
+    pub size: UVec2,
     /// The specific areas of the atlas where each texture can be found
     pub textures: Vec<Rect>,
```

Comment:
Might make sense to turn this into an `URect`?

Reply:
Good point, I'll give it a try and see if this causes conflicts anywhere.

As a side question: Is it a valid use case to specify fractional portions for the textures in a texture atlas? With `URect` I don't think it would be possible anymore. `TextureAtlasLayou::from_grid` is a similar story, now that I look at it.

Reply:
I don't think sub-pixel indexing is a use case needed in the texture atlas code (but I'm not very familiar with that code so I might be wrong), and if a user need for some reason to do some fractional indexing, they can still convert from `UVec2` to `Vec2`. To be consistent some similar changes would also be needed in `TextureAtlasBuilder`, but that can be done in another PR :slightly_smiling_face:

Reply:
I adjusted the types in `TextureAtlasLayout` and `TextureAtlasBuilder` to use "u32" types (`URect`, etc.) as you suggested. Though, because of these changes I also had to update a bit more code than I would have like.

Interactions with `guillotiere` are also a bit odd because [these types use `i32`](https://docs.rs/guillotiere/latest/guillotiere/type.Rectangle.html) instead of `u32`. So theres a few `as i32` here and there.